creditcalc.py

```diff
@@ -9,36 +9,32 @@
 parser.add_argument("--payment", type=float)
 args = parser.parse_args()
 
-if args.interest is None:
-    print("Incorrect parameters.")
-    exit()
+if (args.type != "diff" and args.type != "annuity" or args.interest is None):
+    print("Incorrect parameters. --type can be 'diff' or 'annuity'. --interest is mandatory")
+    exit(-1)
 
-financial_data = [args.type, args.principal, args.periods, args.interest, args.payment]
 nominal_rate = args.interest / 12 / 100
 
-if args.type != 'diff' and args.type != 'annuity':
-    print("Incorrect parameters.")
-if len(financial_data) < 4:
```

Comment:
I suspect this check doesn't do what you expect it does. This length would always be 4 as None types are also counted.